packages/browser-runtime-core/src/open-context-runtime.ts

```diff
@@ -62,4 +64,8 @@ export class OpenContextRuntime implements Runtime {
     this.internalState.setEvaluationListener(listener);
     return prev;
   }
+
+  async getShellPrompt(): Promise<string> {
+    return await this.serviceProvider.getDefaultPrompt();
```

Comment:
Hm… maybe this could still forward to something like `this.internalState.getShellPrompt()`, which then forwards to the service provider? That way we can avoid storing it here just for this

Reply:
Yeah I cleaned up the remnants of the `internalState` now. All places now go to the `serviceProvider` directly. The forwarding didn't really make sense to me anymore since also the user-specific prompt will be implemented and evaluated in `MongoshNodeRepl` which basically has the `serviceProvider` directly already, too. So it made sense to me to get rid of the unnecessary level of indirection.

Reply:
I’m just a bit concerned because this now means that this package has to have knowledge about what the service providers look like, rather than it being just something that it passes through to the `ShellInternalState` constructor

Reply:
Ah very valid point... mh... I've no hard feelings on that one. True, the runtimes just forward that to the internal state...